### ISSUE TITLE
`bun update` doesn't update version number in package.json

### ISSUE THREAD

**Issue by ghost** [oven-sh/bun#5512]
### What version of Bun is running?

1.0.2+37edd5a6e389265738e89265bcbdf2999cb81a49

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

`bun update` should update all dependencies and `bun update {package}` should update the version number of package in package.json

Also `bun update` isn't in the docs.

**Comment by sannajammeh** [comment]
Same issue as Yarn, pnpm... Those two contain the `--latest` flag to modify package.json, I think Bun should include this flag as well.

**Comment by LangLangBart** [comment]
- #4046 implemented `bun update` as equivalent to `npm update`

---

- workaround for the moment with [jq](https://jqlang.github.io/jq/)

```json
  "scripts": {
    "bun:update": "jq '(.dependencies // {}) + (.devDependencies // {}) | keys[]' package.json | xargs bun install"
  }
```

- alternatively: filter out versions without a leading `^` (caret) or `~` (tilde)

```json
  "scripts": {
    "bun:update": "jq '(.dependencies // {}) + (.devDependencies // {}) | to_entries[] | select(.value | test(\"^[~^]\")) | .key' package.json | xargs bun install"
  }
```

**Comment by noClaps** [comment]
Related issue: https://github.com/oven-sh/bun/issues/4312

**Comment by voltuer** [comment]
still nothing since SEPTEMBER?

is this project abandoned?

**Comment by touhidurrr** [comment]
As of `1.1.4`, I am facing problems with:
1. `bun update`
2. `bun update --latest`
3. `bun update -f`
4.  `bun update -f --latest`

Non of these commands update `package.json`.

**Comment by dylan-conway** [comment]
Fixed in #11340
